cmd/advertisement-operator/main.go

```diff
@@ -204,10 +214,14 @@ func main() {
 	if err != nil {
 		os.Exit(1)
 	}
-	wg.Add(3)
-	go advertisementReconciler.CleanOldAdvertisements(c, wg)
+	wg.Add(5)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
As you are deferring, you are causing a deadlock due to circular dependency (with `wg`). You should explicitly call `cancel()` before `wg.Wait`